### PR TITLE
Add get_device for engine_processor

### PR DESCRIPTION
**Pull request by dstrain115**
- Adds get_device() that returns a cirq.Device out of the
device specification.

- Also, cleaned up some documentation that was not using the
engine processor correctly.

### CODE REVIEW

File: cirq/google/engine/engine_processor.py [hunk 1]
```diff
@@ -108,6 +110,15 @@ def get_device_specification(
         else:
             return None
 
+    def get_device(
```
Comment (wcourtney) [inline cirq/google/engine/engine_processor.py:113]: Please add a doc string for this method.

File: cirq/google/engine/engine_processor.py [hunk 2]
```diff
@@ -108,6 +110,21 @@ def get_device_specification(
...
+            self,
+            gate_sets: Iterable[serializable_gate_set.SerializableGateSet]):
```
Comment (dabacon) [inline cirq/google/engine/engine_processor.py:115]: Just a side comment: This probably really shows that the gate set should also be part of the device spec.

Comment (dabacon) [inline cirq/google/engine/engine_processor.py:115]: add return type

Comment (dstrain115) [reply cirq/google/engine/engine_processor.py:115]: Perhaps there is a better way of doing it, but the gate set is part of the device spec.  You just need a way to turn 'fsim' into 'cirq.FSimGate'.  There needs to be a place that can translate the gate ids into cirq objects.

File: cirq/google/engine/engine_processor.py (hunk 1 above)
Comment (dstrain115) [reply cirq/google/engine/engine_processor.py:113]: Done.

File: cirq/google/engine/engine_processor.py (hunk 2 above)
Comment (dstrain115) [reply cirq/google/engine/engine_processor.py]: Done.

**Pull request merged**